apps/src/javalab/JavalabFileHelper.js

```diff
@@ -20,17 +20,53 @@ export const getTabKey = index => `file-${index}`;
 export const fileMetadataForEditor = (sources, isEditingStartSources) => {
   let fileMetadata = {};
   let orderedTabKeys = [];
-
+  let unorderedTabKeys = [];
+  // sources contains file information as key-value pairs.
+  // The key is the file name such as 'MyClass.java'.
+  // The value is a file object with the keys: text (source code),
+  // order (order of the file tab in orderedTabs from 0 to number of files - 1),
+  // isVisible, and isValidation.
+  // The orderOfFiles array will contain the 'order' of each file as they are assigned
+  // their tabKey names (such as 'file-0', 'file-1') which is stored in fileMetadata
```

Comment:
Clarification: does `orderOfFiles` look something like `[1, 0, 3, 2]`, where `1` == index 1 in `unorderedTabKeys`? Since this setup is somewhat complicated it would be good to have an example of the array in the comment so it's clear for future readers :)

Reply:
Yes, that's correct - and I have added detailed comments including an example. Thanks so much Molly!